Testing: Reorganize reaper2 tests #1431

rcarpa opened the pull request:
Split the existing test into two: single vo; and multi vo. Skip the
later in single-vo environments.

The existing test was difficult to follow and understand. I have a
strong feeling that some parts of the tests were not working as
intended. For example: always using rse_names[0] and all_file_names[0]
(and never the second elements of the arrays) in the multi-vo code path
doesn't seem right. Also, exclude_rses was provided an empty list,
which suggests that the author expected include_rses to also take a
list. This is not the case: both take an rse expression string.

Move the relevant multi_vo tests to tests_reaper2.py and merge
duplicate code.

Commit:
Testing: make reaper2 test linear. #1431

Split the existing test into two: single vo; and multi vo. Skip the
later in single-vo environments.

The existing test was difficult to follow and understand. I have a
strong feeling that some parts of the tests were not working as
intended. For example: always using rse_names[0] and all_file_names[0]
(and never the second elements of the arrays) in the multi-vo code path
doesn't seem right. Also, exclude_rses was provided an empty list,
which suggests that the author expected include_rses to also take a
list. This is not the case: both take an rse expression string.

bziemons commented:
I am all for cleaning up :+1:
Can you check if the tests in https://github.com/rucio/rucio/blob/e049b184b55fa4b1304db04d5e88d945d82969e8/lib/rucio/tests/test_multi_vo.py#L1092-L1210 can be included here?

rcarpa commented:
Didn't know about these ones. I see that one of the tests does almost the same thing as the test multi-vo one in test_reaper.py.
Should I move/merge the reaper tests from `test_multi_vo.py` into `test_reaper.py` and skip them in single-vo tests?

bziemons commented:
> Didn't know about these ones. I see that one of the tests does almost the same thing as the test multi-vo one in test_reaper.py.
> Should I move/merge the reaper tests from `test_multi_vo.py` into `test_reaper.py` and skip them in single-vo tests?

sounds like the right thing to do, yes

bari12 commented:
👍 I think that would be good. For MultiVO functionality we do have tests in all the individual test_* files anyway. They are not concentrated to test_multi_vo.py.

bziemons commented:
Please rebase on master.

Commit:
Testing: move multi-vo reaper2 tests to test_reaper2.py. #1431

Code move without functional changes. Some tests are redundant, will
merge them in a later commit.

Commit:
Testing: factorize duplicate code in reaper2 tests. #1431
